v2/tools/generator/internal/astmodel/package_reference.go

```diff
@@ -35,6 +35,15 @@ type PackageReference interface {
 	GroupVersion() (string, string)
 }
 
+type LocalLikePackageReference interface {
```

Comment:
Not sure why this interface is needed.

Reply:
I added the following comment:
```
// LocalLikePackageReference describes a package reference that points to a local package (either a storage package
// or a standard one). It can be used to abstract across the exact package type (storage vs local)
```

It's used like this:
```
	leftLPR, isLeftLocalRef := left.PackageReference.(astmodel.LocalLikePackageReference)
	rightLPR, isRightLocalRef := right.PackageReference.(astmodel.LocalLikePackageReference)
```

In `detect_skipping_properties.go`.

Basically there's code that needs to treat `PackageReference` the same as long as it's one of the two local flavors (Storage | Local).

Is there a better way to do this?

Reply:
I looked into `astmodel.IsExternalPackageReference`, but using that in this situation is awkward because:

1. I need access to `LocalPathPrefix` in addition to `Group`, so I would still need to do 2 casts (one for `LocalPackageReference` and one for `StoragePackageReference`)
2. Performing this check by looking for the "set of things they aren't" is more confusing than a positive assertion. That is: `IsLocalLikePackageReference` should be preferred over `!ExternalPackageReference()` - who knows if/when we add another type and that `!IsExternalPackageReference()` is now a bug.

Additionally, our exisitng `PackageReference` interface shape has always been a peeve of mine, specifically these methods:
```
	// IsPreview returns true if this package reference has a suffix indicating it's a preview
	// release, false otherwise
	IsPreview() bool
	// TryGroupVersion returns the group and version of this reference.
	// Returns true if the reference has a group and version, false otherwise.
	TryGroupVersion() (string, string, bool)
	// GroupVersion returns the group and version of this reference, triggering a panic if they aren't available
	GroupVersion() (string, string)
```

As all of them only make sense for `LocalLikePackageReferences` but we have them on the generic interface even when they don't apply to all that interfaces implementers.

Would you be open to me leaving this as is for now (it definitely makes the checks I need to do easier), and I come back in another PR and resolve some of the above issues as well so that _more_ callers use this interface rather than it just being the one?